### PR TITLE
Cassandra 5 cluster tests

### PR DESCRIPTION
**Pull request by conorbros**
Basic tests for now, going to open a followup with tests similar to the clusterv3/clusterv4 single rack tests.

### CODE REVIEW

File: shotover-proxy/tests/test-configs/cassandra/cluster-v5/topology-dummy-peers.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (rukai) [inline shotover-proxy/tests/test-configs/cassandra/cluster-v5/topology-dummy-peers.yaml:1]: this file is unused

File: shotover-proxy/tests/test-configs/cassandra/cluster-v5/topology-encode.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (rukai) [inline shotover-proxy/tests/test-configs/cassandra/cluster-v5/topology-encode.yaml:1]: this file is unused

Comment (conorbros) [reply shotover-proxy/tests/test-configs/cassandra/cluster-v5/topology-encode.yaml]: I had intended to use them in a followup, but I'll remove from this PR.

**Review by rukai — APPROVED** [review]
lgtm

**Review by cjrolo — APPROVED** [review]

**Pull request merged**